subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/base/KotlinDslBasePlugin.kt

```diff
@@ -47,11 +47,18 @@ abstract class KotlinDslBasePlugin : Plugin<Project> {
 
         apply<KotlinDslCompilerPlugins>()
         addGradleKotlinDslDependencyTo(kotlinArtifactConfigurationNames)
+
+        disableKotlinCompilationAvoidance()
```

Comment:
🤔 Should we move the line before we apply the Kotlin plugin? Although probably plugin doesn't read that property on apply

Reply:
:+1: better safe than sorry